python/tests/test_collections.py

```diff
@@ -210,37 +305,55 @@ def test_collection_document_set_data_uploads_document() -> None:
         COLLECTION_REFERENCE_KEY, COLLECTION_REFERENCE_ID
     )
     gql.collection_document_set.return_value = _collection_document_set_reference(
-        COLECTION_DOCUMNET_KEY
+        COLLECTION_DOCUMNET_KEY
     )
     test_collection = collection(COLLECTION_NAME, _client)
-    document = test_collection.document(COLECTION_DOCUMNET_KEY)
+    document = test_collection.document(COLLECTION_DOCUMNET_KEY)
     assert isinstance(document, NumerousDocument)
     assert document.exists is False
 
     document.set({"test": "test"})
 
+    gql.collection_document_set.assert_called_once_with(
+        COLLECTION_REFERENCE_ID, COLLECTION_DOCUMNET_KEY, BASE64_DOCUMENT_DATA, **KWARGS
+    )
     assert document.exists
 
 
-def test_collection_document_data_decoding_returns_dict() -> None:
+def test_collection_document_get_returns_dict() -> None:
     gql = Mock(GQLClient)
     _client = Client(gql)
     gql.collection_create.return_value = _collection_create_collection_reference(
         COLLECTION_REFERENCE_KEY, COLLECTION_REFERENCE_ID
     )
     gql.collection_document.return_value = _collection_document_reference(
-        COLECTION_DOCUMNET_KEY
+        COLLECTION_DOCUMNET_KEY
     )
     test_collection = collection(COLLECTION_NAME, _client)
-    document = test_collection.document(COLECTION_DOCUMNET_KEY)
+    document = test_collection.document(COLLECTION_DOCUMNET_KEY)
 
     data = document.get()
 
     assert isinstance(document, NumerousDocument)
+    gql.collection_document.assert_has_calls(
+        [
+            call(
+                ORGANIZATION_ID,
+                COLLECTION_REFERENCE_KEY,
+                COLLECTION_DOCUMNET_KEY,
+                **KWARGS,
+            ),
+            call(
+                ORGANIZATION_ID,
+                COLLECTION_REFERENCE_KEY,
+                COLLECTION_DOCUMNET_KEY,
+                **KWARGS,
+            ),
+        ]
+    )
     assert document.exists
-    assert data
-    if data is not None:
-        assert data["test"] == "test"
+    assert data is not None
+    assert data == DOCUMENT_DATA
```

Comment:
You can simplify this as simply `assert data == DOCUMENT_DATA` because if `data == None` that will also fail :)